WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt

```diff
@@ -214,6 +216,10 @@ class WPMainActivityViewModel @Inject constructor(
         }
 
         _mainActions.postValue(actionsList)
+        if (hasLoadedMainActions && actionsList.any { it is AnswerBloggingPromptAction }) {
```

Comment:
I feel the usage of this `hasLoadedMainActions` flag here might be prone to error and difficult to understand. For instance: this works because `loadMainActions` is called exactly once (in the `start` function) before it is then called to update the list (in the `onFabClicked`) function.

I believe the important behavior is calling the tracking when the list was updated as a result of the `onFabClicked` function (meaning the updated actions bottom sheet will be shown to the user), right?

### Suggestions
In this case I believe we could either:

#### Option 1
Add an argument to `loadMainActions` and use that instead of the `hasLoadedMainActions` property, something like:
```kotlin
fun loadMainActions(site: SiteModel?, onFabClicked: Boolean = false) {
    // update the main actions ...
    if (onFabClicked) trackCreateActionsSheetCard(actionsList)
}

fun trackCreateActionsSheetCard(actions: List<MainActionListItem>) {
    if (actions.any { it is AnswerBloggingPromptAction }) {
        analyticsTracker.track(Stat.BLOGGING_PROMPTS_CREATE_SHEET_CARD_VIEWED)
    }
}
```

And call `loadMainActions(site, true)` whenever its called from `onFabClicked`.

#### Option 2
Also create the `trackCreateActionsSheetCard` function like in Option 1 above but instead of coupling it to the `loadMainActions` function, just call it inside `onFabClicked` synchronized with the `loadMainActions` function. 

This would be more complicated as currently `loadMainActions` creates its own Coroutine Job, so it would need to be changed to be a suspending function, and maybe return the value instead of updating the LiveData directly (to make sure `trackCreateActionsSheetCard` would use the updated values), then the job could be created inside `onFabClicked`, here:
https://github.com/wordpress-mobile/WordPress-Android/blob/a9f75001b4cf6e7f3aae6971da35ea4325e4f04e/WordPress/src/main/java/org/wordpress/android/viewmodel/main/WPMainActivityViewModel.kt#L263-L284

By replacing line 276 with:
```kotlin
launch {
    val actionsList = loadMainActions(site)
    _mainActions.postValue(actionsList)
    trackCreateActionsSheetCard(actionsList)
}
```

---

Although this last option makes **tracking** less coupled with **loading data**, I am not sure if this change is worth it so I would suggest **Option 1**. What do you think?